### ISSUE TITLE
Add POST with JSON data example to documentation

### ISSUE THREAD

**Issue by literakl** [sindresorhus/got#1156]
#### What problem are you trying to solve?

I am newbie and I wanted quickly start with Got. I need to find out how to send a POST with JSON body but there was nothing in the documentation at all. I was googling for some tutorial ("got" is very hard to google, "gotjs" would be more unique) until I found some feature request with an example.

#### Describe the feature

Please put the following code at the top of the documentation below the GET example:

got(url, {
	method: 'POST',
	headers: {
		'content-type': 'application/json'
	},
	body: JSON.stringify(body);
});

Another documentation hint: one big document is hard to read, especially when chapters are not numbered. It is hard to find out if the next chapter is the same level, subchapter or higher level. 

Thank you for consideration

#### Checklist

- [x] I have read the documentation and made sure this feature doesn't already exist.

**Comment by sindresorhus** [comment]
There's an example here: https://github.com/sindresorhus/got#json-mode

**Comment by literakl** [comment]
Thank you, I missed it. But this should be one of the first paragraphs, it is hard to locate.

**Comment by sindresorhus** [comment]
Yeah, we can add a POST with JSON example to https://github.com/sindresorhus/got#usage, since it's a common need.

**Comment by literakl** [comment]
Yes, please.
Btw I spent Sunday with some issue that I later identified was related to
Axios. Today it took me ten minutes to fix it with Got.js!

po 13. 4. 2020 v 17:20 odesílatel Sindre Sorhus <notifications@github.com>
napsal:

**Comment by szmarczak** [comment]
> #### Checklist
> 
> - [x] I have read the documentation and made sure this feature doesn't already exist.

<details>
<summary>You haven't read the documentation at all. Me, as a collaborator, by seeing this tick I expect that you are familiar with Got. If you left that unticked, <b>your issue would be still prioritized the same as others</b>. It just gives me a hint where to look at first. </summary>

<br>

> one big document is hard to read

You don't need to read the whole documentation. Just look at what you need. Either you missed the following or didn't bother to check the documentation at all. We've got this at the very top of our readme:

![image](https://user-images.githubusercontent.com/36894700/79139428-92e19a80-7db6-11ea-8172-ec5be0e189a3.png)

> there was nothing in the documentation at all.

I think you didn't bother to read the documentation at all,
because if you made ***a few scrolls down*** then you would notice this:

![image](https://user-images.githubusercontent.com/36894700/79139778-3763dc80-7db7-11ea-8a93-7706df881bba.png)

and this:

![image](https://user-images.githubusercontent.com/36894700/79139848-52365100-7db7-11ea-886a-2cc0387c4310.png)

> It is hard to find out if the next chapter is the same level, subchapter or higher level.

Not saying that the lack of the numeration is a good/bad thing, but if you're familiar with GitHub Markdown then you know that different chapter levels have different font sizes. All GitHub projects I've seen didn't have enumeration. But it's a good thing to consider this change.

</details>

**Comment by szmarczak** [comment]
> Today it took me ten minutes to fix it with Got.js!

Ten minutes is nothing compared to my endless hours spent for tracking "Got" bugs to the Node.js core :wink:

> Yeah, we can add a POST with JSON example to sindresorhus/got#usage, since it's a common need.

I think it'd be best to add a RunKit link to every Got option in the readme.

**Comment by literakl** [comment]
I reported this issue to give you newbie point of view and I wanted to help you. I think that the POST with body example shall be prominent part of the documentation.

> You haven't read the documentation at all. 

I understand your feeling, but you are not right and I feel your comment as offensive. I read the documentation from top to bottom and then I started to search for "body" again. I overlooked this particular occurence, probably because of scrolling too much.

**Comment by szmarczak** [comment]
> I wanted to help you

The intention is greatly appreciated.

> I understand your feeling, but you are not right and I feel your comment as offensive.

You've got every right to. I'm just getting annoyed (perhaps that's because I've memorized parts of the documentation). You are not the first one and the last one who has been having troubles with the readme. I can't blame anybody as we all are trying to do our best.

> I overlooked this particular occurence, probably because of scrolling too much.

***That*** is the feedback we want. Details are precious.

@sindresorhus IMO if we split the readme into different files the readability will be greatly improved.

**Comment by literakl** [comment]
Thank you.

Dne po 13. 4. 2020 22:00 uživatel Szymon Marczak <notifications@github.com>
napsal:

**Comment by christopherreay** [comment]
This issue is just "part" of the node.js http ecosystem. None of the documentation for ANY of the modiules makes it obvious that sending a JSON string to a server without the correct headers doesnt do  much.

Your JSON Mode example, in the docs, does exactly the same thing. It convinces people their JSON should arrive, but actually they will almost certainly receive an error from the server they are posting to, because the content is not JSON.

I dont understand why people dont talk about headers. Are you trying to hide them from people to make stuff more "usable" This issue with JSON headers is.. fpffff. it happens literally thousands of times a day

**Comment by christopherreay** [comment]
nicelibrary

**Comment by szmarczak** [comment]
> Your JSON Mode example, in the docs, does exactly the same thing. It convinces people their JSON should arrive, but actually they will almost certainly receive an error from the server they are posting to, because the content header is not set, so whatever server will not interpret it as such;

You're yet another person who doesn't read the documentation. 

See https://github.com/sindresorhus/got/#json

https://github.com/sindresorhus/got/blob/c56c33ad3fb70b0be26c4469accdc0ffc7de6f8c/source/core/index.ts#L922-L926

Locking this issue to prevent spreading misinformation.
